pkg/controller/storagecluster/reconcile.go

```diff
@@ -119,3 +215,85 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
 		},
 	}
 }
+
+// setProgressingCondition sets the ProgressingCondition to True and other conditions to
+// false or Unknown. Used when we are just starting to reconcile, and there are no existing
+// conditions.
+func setProgressingCondition(conditions *[]conditionsv1.Condition, reason string, message string) {
```

Comment:
This is being duplicated from the ocsinitialization controller. Can you put this and the functions below, into a new package, and call them that way.